File: AppShoeStoreBE/src/main/java/com/example/AppShoeStoreBE/Service/User/UserService.java
package com.example.AppShoeStoreBE.Service.User;

import com.example.AppShoeStoreBE.Entity.User;
import com.example.AppShoeStoreBE.Request.User.CreateAccountRequest;
import com.example.AppShoeStoreBE.Request.User.LoginAccountRequest;
import com.example.AppShoeStoreBE.Util.ApiResponse;
import org.springframework.stereotype.Service;

@Service
public interface UserService {

    // create account
    ApiResponse<User> CreateAccount(CreateAccountRequest createAccountRequest);

    // login account
    ApiResponse<User> LoginAccount(LoginAccountRequest loginRequest);
}

File: AppShoeStore/app/src/main/java/com/example/appshoestore/Service/UserService.kt
package com.example.appshoestore.Service

import com.example.appshoestore.Model.Products
import com.example.appshoestore.Model.User
import com.example.appshoestore.Request.CreateAccountRequest
import com.example.appshoestore.Request.LoginAccountRequest
import com.example.appshoestore.Util.ApiResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface UserService {

    // create account
    @POST("user/createAccount")
    suspend fun createAccount(@Body request: CreateAccountRequest): Response<ApiResponse<User>>

    // login account
    @POST("user/loginAccount")
    suspend fun loginAccount(@Body request: LoginAccountRequest): Response<ApiResponse<User>>
}
